packages/core/test/dispatcher.test.ts

```diff
@@ -117,9 +117,47 @@ describe('dispatch:', () => {
 				b: 1,
 			})
 		})
+
+		it('should pass the meta object as the third param', () => {
+			type CountState = {
+				count: number
+				meta: any
+			}
+
+			const count = {
+				state: {
+					count: 0,
+					meta: null,
+				} as CountState,
+				reducers: {
+					add: (
+						state: CountState,
+						payload: number,
+						meta?: Record<string, any>
+					): CountState => {
+						return {
+							count: state.count + payload,
+							meta,
+						}
+					},
+				},
+			}
+
+			const store = init({
+				models: { count },
+			})
+
+			// TODO: type of `add` inferred incorrectly.
```

Comment:
There is an issue(maybe) with [typescript](https://github.com/microsoft/TypeScript/issues/42072). So the optional `meta` isn't supported. You can also click [playground](https://www.typescriptlang.org/play?ts=4.2.0-dev.20201221#code/KYDwDg9gTgLgBASwHY2FAZgQwMbDgQWxgQiQB4AVABUwE8AbCTAEzgF45MlaAaOCgLLAYmdp24A+OAG8AUHAVwwdRiwD8ALn40GTZvMUBbYZk38hI2QF9ZsmLTB5qKvWKQBXQwCM0cAD5w7kjMwOjIwPr2juYmYgBKwNjQzGQAzjBQyADmfFy0UgFBIWFIEbZReAAiCKnKMNgAFmgAYtAAQphQlDqqrBwAbhAIzHyCsQNDzFIcANrOuiyjFpgAunCgqMGpcDODw3x7zCsGCmpwABTnAJTsUoTEpGSHB5MSNwBkMoipAKLo6IkYFosPRUngbIo4FoxiJ1iBNsxtocTnAzkVQuFWBtgFttC4WCizpdlAtmGZ5r0bmw7kQSOQKXoXsM3nBPnJIZCan8AURgZhQcAURDFFpifjmNCenoqTSHvSpYs4IcWWyUZzfv9AXyBUKUVoZuiShE+IbMWtsbi5gqRjERMdIUTziTeuTrXxjCJycsZQRaY8GYqYZgVTI1YouZreXAQWDdZCtKbSlj4TjEbbMISLk7xZLxe6TF6TD77nTunn0yH2RyFBGeUDo-zY5DhQpRdnSbnSfmRNDvbdfXKy12Kx9Q9Waxq69qm4obLIAPQAKkX8kXcAqcA6UDEl3bLs7lP7Jf9bqVr1HVYn3K1DYFAG5rDcArvnXoD4y4B7MFoEkkoCl0kyJAcnEfJiz9eVy1-ZIyAAOngiRKxReDYIfKwrlXec7AcPAtzEapakweomigVooC3IdeiWEwJFsehhDgLxOi0LcHwXZdVzgej4CYqBRVfFh30VL8f0SGDAOyXJJHAwcAxtZV-AHUs5L4aD-zSDJJNAxCUQANU6BBMC8ei4AAcl40zvkCMFWB8dBoDwHxsk4VJUgQLIk1g84ACYABYAFZfIwyEfnAQEIjgbzOCgLJPBxGBUj4Lx3HgLIIHgAAGLzvP8wLgvDJAwBS2CEpAHyAHY+AATiuLR8CQaLYuMFBo2gMyBOYSyAHdMG2JB0qUKAIH6YYIlgzDZF4642KXFc4DXDcOgALx3Pc9Fdcsv0LEQZOU08gyQ9VryjGNgDQibFswFbZhU9NzRTS1njPYZ7UUR1dseJ7lQvb5jvrU64BbKF0zhBEkUmTNEwii00zkzMxVJDbSQ+yDh2+1kx3HWsb1OuMRSzDqhOYFHKI-dHVXHX7I3+xtBWbPUdihm0mfusGdluoNXtOAnxSRqjPwLXsiyPCDScDZZDqxyccdpvHW0CYIMSTUHU22IN4bWwS8WHLahZ2kXZP2iWLzDK9qenOnZwZhHeiJ7tv3TEmOeNjHLw5bGTtl+nZG4xirpYq62KYpbpsmq7zgARiuB9g8jvhpDM4PTK0CPAej2QgA) for more details.

Reply:
I fixed that.